protocol/src/format.rs

```diff
@@ -12,89 +12,230 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use std::fmt;
-use std::mem;
+use std::{fmt, str::FromStr};
+
+use crate::translate;
+use crate::translate::*;
+
+use serde::Deserialize;
+
+pub use crate::format::color::*;
+
+const LEGACY_CHAR: char = '§';
 
 #[derive(Debug, Clone)]
-pub enum Component {
-    Text(TextComponent),
+pub struct Component {
```

Comment:
Are you sure, it's correct to transform Component into a struct? Because how did you plan to support clickable/hoverable/web link components?

Reply:
I would make the rest ready for review tomorrow.